twine-components/src/thermal/hx.rs

```diff
@@ -0,0 +1,108 @@
+#![allow(dead_code)]
+
+use std::ops::Deref;
+
+use twine_core::constraint::{Constrained, ConstraintError, NonNegative, UnitInterval};
+use uom::si::{f64::ThermalConductance, thermal_conductance::watt_per_kelvin};
+
+/// A fluid capacitance rate.
+#[derive(Debug, Clone, Copy, PartialEq, PartialOrd)]
+struct CapacitanceRate(Constrained<ThermalConductance, NonNegative>);
+
+impl CapacitanceRate {
+    /// Constructs a `CapacitanceRate`.
+    ///
+    /// # Errors
+    ///
+    /// Fails if the value is negative.
+    fn new(value: ThermalConductance) -> Result<Self, ConstraintError> {
+        Ok(Self(NonNegative::new(value)?))
+    }
+}
+
+impl TryFrom<ThermalConductance> for CapacitanceRate {
+    type Error = ConstraintError;
+
+    fn try_from(value: ThermalConductance) -> Result<Self, Self::Error> {
+        CapacitanceRate::new(value)
+    }
+}
+
+impl Deref for CapacitanceRate {
+    type Target = ThermalConductance;
+
+    fn deref(&self) -> &Self::Target {
+        self.0.as_ref()
+    }
+}
+
+/// The capacity ratio of two fluids in a heat exchanger.
+#[derive(Debug, Clone, Copy, PartialEq, PartialOrd)]
+struct CapacityRatio(Constrained<f64, UnitInterval>);
```

Comment:
I would really prefer that this be `Constrained<Ratio, UnitInterval>`, but Ratio doesn't implement `num_traits::One`, so I can't use it here. I'm very open to ideas here if you have some.